packages/block-editor/src/components/link-control/index.js

```diff
@@ -202,7 +185,7 @@ function LinkControl( {
 			<div className="block-editor-link-control__popover-inner">
 				<div className="block-editor-link-control__search">
 
-					{ ( ! isEditingLink && currentLink ) && (
+					{ ( ! isEditingLink ) && (
```

Comment:
If we don't have a link then we don't want to show this UI interface. Just curious as to why you feel it's safe to remove this now?

Reply:
the boolean seemed explicit enough for me. If we do add the check for the  value here, we should also add it to force showing the input if the  value is  empty which was not the case.